File: extensions/spark/kyuubi-spark-connector-tpcds/src/main/java/io/trino/tpcds/row/WebSiteRow.java
package io.trino.tpcds.row;

    import io.trino.tpcds.type.Address;
    import io.trino.tpcds.type.Decimal;
    import org.apache.kyuubi.spark.connector.tpcds.row.KyuubiTPCDSTableRowWithNulls;

    import java.util.List;

    import static com.google.common.collect.Lists.newArrayList;
    import static io.trino.tpcds.generator.WebSiteGeneratorColumn.WEB_ADDRESS_CITY;
    import static io.trino.tpcds.generator.WebSiteGeneratorColumn.WEB_ADDRESS_COUNTRY;
    import static io.trino.tpcds.generator.WebSiteGeneratorColumn.WEB_ADDRESS_COUNTY;
    import static io.trino.tpcds.generator.WebSiteGeneratorColumn.WEB_ADDRESS_GMT_OFFSET;
    import static io.trino.tpcds.generator.WebSiteGeneratorColumn.WEB_ADDRESS_STATE;
    import static io.trino.tpcds.generator.WebSiteGeneratorColumn.WEB_ADDRESS_STREET_NAME1;
    import static io.trino.tpcds.generator.WebSiteGeneratorColumn.WEB_ADDRESS_STREET_NUM;
    import static io.trino.tpcds.generator.WebSiteGeneratorColumn.WEB_ADDRESS_STREET_TYPE;
    import static io.trino.tpcds.generator.WebSiteGeneratorColumn.WEB_ADDRESS_SUITE_NUM;
    import static io.trino.tpcds.generator.WebSiteGeneratorColumn.WEB_ADDRESS_ZIP;
    import static io.trino.tpcds.generator.WebSiteGeneratorColumn.WEB_CLASS;
    import static io.trino.tpcds.generator.WebSiteGeneratorColumn.WEB_CLOSE_DATE;
    import static io.trino.tpcds.generator.WebSiteGeneratorColumn.WEB_COMPANY_ID;
    import static io.trino.tpcds.generator.WebSiteGeneratorColumn.WEB_COMPANY_NAME;
    import static io.trino.tpcds.generator.WebSiteGeneratorColumn.WEB_MANAGER;
    import static io.trino.tpcds.generator.WebSiteGeneratorColumn.WEB_MARKET_CLASS;
    import static io.trino.tpcds.generator.WebSiteGeneratorColumn.WEB_MARKET_DESC;
    import static io.trino.tpcds.generator.WebSiteGeneratorColumn.WEB_MARKET_ID;
    import static io.trino.tpcds.generator.WebSiteGeneratorColumn.WEB_MARKET_MANAGER;
    import static io.trino.tpcds.generator.WebSiteGeneratorColumn.WEB_NAME;
    import static io.trino.tpcds.generator.WebSiteGeneratorColumn.WEB_OPEN_DATE;
    import static io.trino.tpcds.generator.WebSiteGeneratorColumn.WEB_REC_END_DATE_ID;
    import static io.trino.tpcds.generator.WebSiteGeneratorColumn.WEB_REC_START_DATE_ID;
    import static io.trino.tpcds.generator.WebSiteGeneratorColumn.WEB_SITE_ID;
    import static io.trino.tpcds.generator.WebSiteGeneratorColumn.WEB_SITE_SK;
    import static io.trino.tpcds.generator.WebSiteGeneratorColumn.WEB_TAX_PERCENTAGE;
    import static java.lang.String.format;

public class WebSiteRow
    extends KyuubiTPCDSTableRowWithNulls
{
  private final long webSiteSk;
  private final String webSiteId;
  private final long webRecStartDateId;
  private final long webRecEndDateId;
  private final String webName;
  private final long webOpenDate;
  private final long webCloseDate;
  private final String webClass;
  private final String webManager;
  private final int webMarketId;
  private final String webMarketClass;
  private final String webMarketDesc;
  private final String webMarketManager;
  private final int webCompanyId;
  private final String webCompanyName;
  private final Address webAddress;
  private final Decimal webTaxPercentage;

  public WebSiteRow(long nullBitMap,
      long webSiteSk,
      String webSiteId,
      long webRecStartDateId,
      long webRecEndDateId,
      String webName,
      long webOpenDate,
      long webCloseDate,
      String webClass,
      String webManager,
      int webMarketId,
      String webMarketClass,
      String webMarketDesc,
      String webMarketManager,
      int webCompanyId,
      String webCompanyName,
      Address webAddress,
      Decimal webTaxPercentage)
  {
    super(nullBitMap, WEB_SITE_SK);
    this.webSiteSk = webSiteSk;
    this.webSiteId = webSiteId;
    this.webRecStartDateId = webRecStartDateId;
    this.webRecEndDateId = webRecEndDateId;
    this.webName = webName;
    this.webOpenDate = webOpenDate;
    this.webCloseDate = webCloseDate;
    this.webClass = webClass;
    this.webManager = webManager;
    this.webMarketId = webMarketId;
    this.webMarketClass = webMarketClass;
    this.webMarketDesc = webMarketDesc;
    this.webMarketManager = webMarketManager;
    this.webCompanyId = webCompanyId;
    this.webCompanyName = webCompanyName;
    this.webAddress = webAddress;
    this.webTaxPercentage = webTaxPercentage;
  }

  public String getWebName()
  {
    return webName;
  }

  public long getWebOpenDate()
  {
    return webOpenDate;
  }

  public long getWebCloseDate()
  {
    return webCloseDate;
  }

  public String getWebClass()
  {
    return webClass;
  }

  public String getWebManager()
  {
    return webManager;
  }

  public int getWebMarketId()
  {
    return webMarketId;
  }

  public String getWebMarketClass()
  {
    return webMarketClass;
  }

  public String getWebMarketDesc()
  {
    return webMarketDesc;
  }

  public String getWebMarketManager()
  {
    return webMarketManager;
  }

  public int getWebCompanyId()
  {
    return webCompanyId;
  }

  public String getWebCompanyName()
  {
    return webCompanyName;
  }

  public Address getWebAddress()
  {
    return webAddress;
  }

  public Decimal getWebTaxPercentage()
  {
    return webTaxPercentage;
  }

  public long getWebSiteSk() {
    return webSiteSk;
  }

  public String getWebSiteId() {
    return webSiteId;
  }

  public long getWebRecStartDateId() {
    return webRecStartDateId;
  }

  public long getWebRecEndDateId() {
    return webRecEndDateId;
  }

  @Override public Object[] values() {
    return new Object[] {
        getOrNullForKey(webSiteSk, WEB_SITE_SK),
        getOrNull(webSiteId, WEB_SITE_ID),
        getDateOrNullFromJulianDays(webRecStartDateId, WEB_REC_START_DATE_ID),
        getDateOrNullFromJulianDays(webRecEndDateId, WEB_REC_END_DATE_ID),
        getOrNull(webName, WEB_NAME),
        getOrNullForKey(webOpenDate, WEB_OPEN_DATE),
        getOrNullForKey(webCloseDate, WEB_CLOSE_DATE),
        getOrNull(webClass, WEB_CLASS),
        getOrNull(webManager, WEB_MANAGER),
        getOrNull(webMarketId, WEB_MARKET_ID),
        getOrNull(webMarketClass, WEB_MARKET_CLASS),
        getOrNull(webMarketDesc, WEB_MARKET_DESC),
        getOrNull(webMarketManager, WEB_MARKET_MANAGER),
        getOrNull(webCompanyId, WEB_COMPANY_ID),
        getOrNull(webCompanyName, WEB_COMPANY_NAME),
        getOrNull(webAddress.getStreetNumber(), WEB_ADDRESS_STREET_NUM),
        getOrNull(webAddress.getStreetName(), WEB_ADDRESS_STREET_NAME1),
        getOrNull(webAddress.getStreetType(), WEB_ADDRESS_STREET_TYPE),
        getOrNull(webAddress.getSuiteNumber(), WEB_ADDRESS_SUITE_NUM),
        getOrNull(webAddress.getCity(), WEB_ADDRESS_CITY),
        getOrNull(webAddress.getCounty(), WEB_ADDRESS_COUNTY),
        getOrNull(webAddress.getState(), WEB_ADDRESS_STATE),
        getOrNull(format("%05d", webAddress.getZip()), WEB_ADDRESS_ZIP),
        getOrNull(webAddress.getCountry(), WEB_ADDRESS_COUNTRY),
        getOrNull(webAddress.getGmtOffset(), WEB_ADDRESS_GMT_OFFSET),
        getOrNull(webTaxPercentage, WEB_TAX_PERCENTAGE)
    };
  }
}

File: extensions/spark/kyuubi-spark-connector-tpcds/src/main/scala/io/trino/tpcds/row/WebSiteRow.scala
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package io.trino.tpcds.row

import io.trino.tpcds.generator.WebSiteGeneratorColumn._
import io.trino.tpcds.`type`.{Address, Decimal}

import java.util.{List => JList}

import org.apache.spark.sql.catalyst.expressions.GenericInternalRow
import org.apache.spark.sql.catalyst.InternalRow

class WebSiteRow(
                  nullBitMap: Long,
                  private val webSiteSk: Long,
                  private val webSiteId: String,
                  private val webRecStartDateId: Long,
                  private val webRecEndDateId: Long,
                  private val webName: String,
                  private val webOpenDate: Long,
                  private val webCloseDate: Long,
                  private val webClass: String,
                  private val webManager: String,
                  private val webMarketId: Int,
                  private val webMarketClass: String,
                  private val webMarketDesc: String,
                  private val webMarketManager: String,
                  private val webCompanyId: Int,
                  private val webCompanyName: String,
                  private val webAddress: Address,
                  private val webTaxPercentage: Decimal
  ) extends TableRowWithNulls(nullBitMap, WEB_SITE_SK)
  with KyuubiTableRowWithNulls {

    override val nullBitMapInternal = nullBitMap
    override val firstColumnInternal = WEB_SITE_SK

  def getWebName: String = webName
  def getWebOpenDate: Long = webOpenDate
  def getWebCloseDate: Long = webCloseDate
  def getWebClass: String = webClass
  def getWebManager: String = webManager
  def getWebMarketId: Int = webMarketId
  def getWebMarketClass: String = webMarketClass
  def getWebMarketDesc: String = webMarketDesc
  def getWebMarketManager: String = webMarketManager
  def getWebCompanyId: Int = webCompanyId
  def getWebCompanyName: String = webCompanyName
  def getWebAddress: Address = webAddress
  def getWebTaxPercentage: Decimal = webTaxPercentage
  def getWebSiteSk: Long = webSiteSk
  def getWebSiteId: String = webSiteId
  def getWebRecStartDateId: Long = webRecStartDateId
  def getWebRecEndDateId: Long = webRecEndDateId

  override def getValues: JList[String] = throw new UnsupportedOperationException()

  def internalRow: InternalRow =
    new GenericInternalRow(
      Array(
    getLongOrNull(webSiteSk, WEB_SITE_SK),
    getStringOrNullInternal(webSiteId, WEB_SITE_ID),
    getDateOrNullFromJulianDays(webRecStartDateId, WEB_REC_START_DATE_ID),
    getDateOrNullFromJulianDays(webRecEndDateId, WEB_REC_END_DATE_ID),
    getStringOrNullInternal(webName, WEB_NAME),
    getLongOrNull(webOpenDate, WEB_OPEN_DATE),
    getLongOrNull(webCloseDate, WEB_CLOSE_DATE),
    getStringOrNullInternal(webClass, WEB_CLASS),
    getStringOrNullInternal(webManager, WEB_MANAGER),
    getIntOrNull(webMarketId, WEB_MARKET_ID),
    getStringOrNullInternal(webMarketClass, WEB_MARKET_CLASS),
    getStringOrNullInternal(webMarketDesc, WEB_MARKET_DESC),
    getStringOrNullInternal(webMarketManager, WEB_MARKET_MANAGER),
    getIntOrNull(webCompanyId, WEB_COMPANY_ID),
    getStringOrNullInternal(webCompanyName, WEB_COMPANY_NAME),
    getStringOrNullInternal(webAddress.getStreetNumber.toString, WEB_ADDRESS_STREET_NUM),
    getStringOrNullInternal(webAddress.getStreetName(), WEB_ADDRESS_STREET_NAME1),
    getStringOrNullInternal(webAddress.getStreetType(), WEB_ADDRESS_STREET_TYPE),
    getStringOrNullInternal(webAddress.getSuiteNumber(), WEB_ADDRESS_SUITE_NUM),
    getStringOrNullInternal(webAddress.getCity(), WEB_ADDRESS_CITY),
    getStringOrNullInternal(webAddress.getCounty(), WEB_ADDRESS_COUNTY),
    getStringOrNullInternal(webAddress.getState(), WEB_ADDRESS_STATE),
    getStringOrNullInternal(f"${webAddress.getZip}%05d", WEB_ADDRESS_ZIP),
    getStringOrNullInternal(webAddress.getCountry(), WEB_ADDRESS_COUNTRY),
    getDecimalOrNull(webAddress.getGmtOffset(), WEB_ADDRESS_GMT_OFFSET, 5, 2),
    getDecimalOrNull(webTaxPercentage, WEB_TAX_PERCENTAGE, 5, 2)))
}
